contracts/HeapOrdering.sol

```diff
@@ -0,0 +1,318 @@
+// SPDX-License-Identifier: GNU AGPLv3
+pragma solidity ^0.8.0;
+
+library HeapOrdering {
+    struct Account {
+        address id; // The address of the account.
+        uint256 value; // The value of the account.
+    }
+
+    struct HeapArray {
+        Account[] accounts; // All the accounts.
+        uint256 size; // The size of the heap portion of the structure, should be less than accounts length, the rest is an unordered array.
```

Comment:
So this would kind of be a bit of a hacky non-standard thing, but I believe that gas savings are possible by converting size to a uint96 and packing it with id into the accounts struct at accounts[0]. This way, size can share a storage slot with address. Maybe worth investigating the possibility?

Reply:
See #30